kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,446 @@
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetStringListIsCalledWith("testKey-comma-seperated");
+		thenNoExceptionsOccured();
+		thenStringListResultEquals(Arrays.asList("commaSeperated1", "commaSeperated2", "commaSeperated3"));
+	}
+
+	@Test
+	public void getStringListShouldThrow() {
+		givenMapWith("testKey-comma-seperated", null);
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetStringListIsCalledWith("testKey-comma-seperated");
+		thenANoSuchElementExceptionOccured();
+	}
+
+	@Test
+	public void getStringListShouldThrowWithNonExistantKey() {
+		givenMapWith("testKey-comma-seperated", null);
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetStringListIsCalledWith("testKey-comma-seperated-not-existant");
+		thenANoSuchElementExceptionOccured();
+	}
+
+	@Test
+	public void getStringListShouldReturnStringListWithOneItem() {
+		givenMapWith("testKey1", "testString1");
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetStringListIsCalledWith("testKey1");
+		thenNoExceptionsOccured();
+		thenStringListResultEquals(Arrays.asList("testString1"));
+	}
+
+	@Test
+	public void getStringListShouldReturnStringListWithNull() {
+		givenMapWith("testKeyNull", null);
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetStringListIsCalledWith("testKeyNull");
+		thenANoSuchElementExceptionOccured();
+	}
+
+	@Test
+	public void getStringListShouldReturnOptionalStringListWithCommaSeperatedString() {
+		givenMapWith("testKey-comma-seperated", "commaSeperated1,commaSeperated2,commaSeperated3");
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetStringListOptIsCalledWith("testKey-comma-seperated");
+		thenNoExceptionsOccured();
+		thenOptionalResultEquals(Optional.of(Arrays.asList("commaSeperated1", "commaSeperated2", "commaSeperated3")));
+	}
+
+	@Test
+	public void getStringListShouldReturnOptionalStringListtWithCommaSeperatedStringMalformed() {
+		givenMapWith("testKey-comma-seperated",
+				", , ,,,,commaSeperated1, , , ,,,,,commaSeperated2,,,, ,, ,,commaSeperated3,, , ,,,, ,");
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetStringListOptIsCalledWith("testKey-comma-seperated");
+		thenNoExceptionsOccured();
+		thenOptionalResultEquals(Optional.of(Arrays.asList("commaSeperated1", "commaSeperated2", "commaSeperated3")));
+	}
+
+	@Test
+	public void getStringListShouldReturnOptionalStringListWithString() {
+		givenMapWith("testKey1", "testString1");
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetStringListOptIsCalledWith("testKey1");
+		thenNoExceptionsOccured();
+		thenOptionalResultEquals(Optional.of(Arrays.asList("testString1")));
+	}
+
+	@Test
+	public void getStringListShouldThrowWithNull() {
+		givenMapWith("testKeyNull", null);
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetStringListOptIsCalledWith("testKeyNull");
+		thenANoSuchElementExceptionOccured();
+	}
+
+	@Test
+	public void getStringListShouldThrowWithNoKey() {
```

Comment:
This isn't our expectation. `getOptStringList` should return `Optional.empty` if the key is missing.

```suggestion
	public void getStringListShouldWorkWithMissingKey() {
```